src/ranged.cpp

```diff
@@ -1667,3 +1669,117 @@ void drop_or_embed_projectile( const dealt_projectile_attack &attack )
     }
 }
 
+double player::gun_value( const item &weap, long ammo ) const
+{
+    // TODO: Mods
+    // TODO: Allow using a specified type of ammo rather than default
+    if( weap.type->gun.get() == nullptr ) {
+        return 0.0;
+    }
+
+    if( ammo <= 0 ) {
+        return 0.0;
+    }
+
+    const islot_gun& gun = *weap.type->gun.get();
```

Comment:
Calling `get()` isn't needed, `unique_ptr` has an overload `operator*`, so `*weap.type->gun` is enough.